tests/unittests/main.c

```diff
@@ -25,6 +25,11 @@ int main(void)
 {
     test_utils_interactive_sync();
 
+#ifdef MODULE_ZTIMER
```

Comment:
I think this could be tightened to '#if MODULE_ZTIMER_USEC || MODULE_ZTIMER_MSEC || MODULE_ZTIMER_SEC'. Some tests (e.g., ztimer_mock tests) don't expect those to be available, and all that ztimer_init() does is set up the default usec,msec,sec clocks.

Reply:
is working -> got that local -> will push when i know what to do with the casts -> i will also do a rebase then since there is something else fixed in master that helps with building the unittests for this